### PR TITLE
Feat:versioning API

### PR DESCRIPTION
**Pull request by Freeman-kuch**
- ​Versioned all API endpoints 
- pluralized endpoints from singular references to plural 

## Description

​- ​Versioned all API endpoints from `/api/admin/product` to `/api/v1/admin/product`
- pluralized endpoints form singular references to plural `/api/v1/admin/products` and so on

​
## Motivation and Context

​Boss @zxenonx  ordered it be done!

## How Has This Been Tested?

​- all refactored endpoints were tested using postman

## Screenshots (if appropriate - Postman, etc):

Screenshot (28)
Screenshot (29)
Screenshot (30)
Screenshot (31)

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Freeman-kuch** [comment]
@zxenonx

File: super_admin_1/products/routes.py [hunk 1]
```diff
@@ -551,7 +551,7 @@ def temporary_delete(user_id, product_id):
 
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("approve_product/<product_id>", methods=["PATCH"])
+@product.route("approve-products/<product_id>", methods=["PATCH"])
 @admin_required(request=request)
 def approve_product(user_id, product_id):  #FOR SANCTIONED PRODUCTS
```
[the remark this answers is not in the document]
Comment (zxenonx) [reply super_admin_1/products/routes.py:556]: docstring missing

File: super_admin_1/products/routes.py [hunk 2]
```diff
@@ -644,7 +644,7 @@ def approve_product(user_id, product_id):  #FOR SANCTIONED PRODUCTS
 
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("delete_product/<product_id>", methods=["DELETE"])
+@product.route("delete-products/<product_id>", methods=["DELETE"])
 @admin_required(request=request)
 def permanent_delete(user_id, product_id):
```
[the remark this answers is not in the document]
Comment (zxenonx) [reply super_admin_1/products/routes.py:649]: docstring missing

File: super_admin_1/shop/routes.py [hunk 3]
```diff
@@ -161,7 +161,7 @@ def check_status(shop):
         return jsonify({"error": "Internal Server Error", "message": str(e)}), 500
 
 
-@shop.route("/all/total_sales", methods=["POST"])
+@shop.route("/all/total-sales", methods=["POST"])
 # @cache.cached(timeout=5)
 @admin_required(request=request)
 def total_shop_sales(user_id) -> defaultdict:
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:167]: type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 4]
```diff
@@ -458,7 +458,7 @@ def ban_vendor(user_id, vendor_id):
 # WORKS - Documented
 
 
-@shop.route("/banned_vendors", methods=["GET"])
+@shop.route("/banned-vendors", methods=["GET"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:461]:
```suggestion
@shop.route("/vendors?status=banned", methods=["GET"])
```

File: super_admin_1/shop/routes.py [hunk 5]
```diff
@@ -458,7 +458,7 @@ def ban_vendor(user_id, vendor_id):
...
 @admin_required(request=request)
 def get_banned_vendors(user_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:463]: type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 6]
```diff
@@ -503,7 +503,7 @@ def get_banned_vendors(user_id):
 
 # Define a route to unban a vendor
 # WORKS - Documented
-@shop.route("/unban_vendor/<vendor_id>", methods=["PUT"])
+@shop.route("/unban-vendors/<vendor_id>", methods=["PUT"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:506]:
```suggestion
@shop.route("/vendors/<vendor_id>/unban", methods=["PUT"])
```

File: super_admin_1/shop/routes.py [hunk 7]
```diff
@@ -503,7 +503,7 @@ def get_banned_vendors(user_id):
...
 @admin_required(request=request)
 def unban_vendor(user_id, vendor_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:508]: type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 8]
```diff
@@ -662,7 +662,7 @@ def restore_shop(user_id, shop_id):
 
 
 # WORKS - Documented
-@shop.route("delete_shop/<shop_id>", methods=["PATCH"])
+@shop.route("delete-shop/<shop_id>", methods=["PATCH"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:665]: check soft delete suggestion above

File: super_admin_1/shop/routes.py [hunk 9]
```diff
@@ -662,7 +662,7 @@ def restore_shop(user_id, shop_id):
...
 @admin_required(request=request)
 def delete_shop(user_id, shop_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:667]: type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 10]
```diff
@@ -729,7 +729,7 @@ def delete_shop(user_id, shop_id):
 
 # delete shop object permanently out of the DB
 # works - Documented
-@shop.route("delete_shop/<shop_id>", methods=["DELETE"])
+@shop.route("delete-shop/<shop_id>", methods=["DELETE"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:732]:
```suggestion
@shop.route("shops/<shop_id>", methods=["DELETE"])
```

File: super_admin_1/shop/routes.py [hunk 11]
```diff
@@ -729,7 +729,7 @@ def delete_shop(user_id, shop_id):
...
 @admin_required(request=request)
 def perm_del(user_id, shop_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:734]: - permanent_delete

- type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 12]
```diff
@@ -774,7 +774,7 @@ def perm_del(user_id, shop_id):
 # Define a route to get all temporarily deleted vendors
 
 
-@shop.route("/temporarily_deleted_vendors", methods=["GET"])
+@shop.route("/temporarily-deleted-vendors", methods=["GET"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:777]:
```suggestion
@shop.route("/vendors?status=soft-deleted", methods=["GET"])
```

File: super_admin_1/shop/routes.py [hunk 13]
```diff
@@ -774,7 +774,7 @@ def perm_del(user_id, shop_id):
...
 @admin_required(request=request)
 def get_temporarily_deleted_vendors(user_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:779]: type hint and docstring missing

File: super_admin_1/shop/routes.py [hunk 14]
```diff
@@ -832,7 +832,7 @@ def get_temporarily_deleted_vendors(user_id):
 
 
 # Define a route to get details of a temporarily deleted vendor based on his/her ID
-@shop.route("/temporarily_deleted_vendor/<string:vendor_id>",  methods=["GET"])
+@shop.route("/temporarily-deleted-vendor/<string:vendor_id>",  methods=["GET"])
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:835]:
```suggestion
@shop.route("/vendors/<string:vendor_id>?status=soft-deleted",  methods=["GET"])
```

File: super_admin_1/shop/routes.py [hunk 15]
```diff
@@ -832,7 +832,7 @@ def get_temporarily_deleted_vendors(user_id):
...
 # WORKS - Documented
 @admin_required(request=request)
 def get_temporarily_deleted_vendor(user_id, vendor_id):
```
Comment (zxenonx) [inline super_admin_1/shop/routes.py:838]: type hint and docstring missing

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @Freeman-kuch 
I left some comments.

File: super_admin_1/products/routes.py [hunk 16]
```diff
@@ -364,7 +364,7 @@ def to_sanction_product(user_id, product_id):
 
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("/product_statistics", methods=["GET"])
+@product.route("/products-statistics", methods=["GET"])
 @admin_required(request=request)
 def get_product_statistics(user_id):
```
[the remark this answers is not in the document]
Comment (Freeman-kuch) [reply super_admin_1/products/routes.py:369]: where ? for the user_id?

[the remark this answers is not in the document]
Comment (zxenonx) [reply super_admin_1/products/routes.py:369]: Yes

[the remark this answers is not in the document]
Comment (zxenonx) [reply super_admin_1/products/routes.py:369]: Type hints for all function parameters

File: super_admin_1/shop/routes.py (hunk 4 above)
Comment (Freeman-kuch) [reply super_admin_1/shop/routes.py:461]: this route would most likely be edprecated, its not actually being used

the shops/all endpoint handles it

File: super_admin_1/products/routes.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (Freeman-kuch) [reply super_admin_1/products/routes.py:369]: ok boss

File: super_admin_1/products/routes.py [hunk 17]
```diff
@@ -399,7 +399,7 @@ def get_product_statistics(user_id):
         ), 400
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("/restore_product/<product_id>", methods=["PATCH"])
+@product.route("/restore-products/<product_id>", methods=["PATCH"])
```
[the remark this answers is not in the document]
Comment (Freeman-kuch) [reply super_admin_1/products/routes.py:402]: alriight boss

File: super_admin_1/products/routes.py [hunk 18]
```diff
@@ -470,7 +470,7 @@ def to_restore_product(user_id, product_id):  #FOR TEMPORARILY DELETED PRODUCT
 
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("delete_product/<product_id>", methods=["PATCH"])
+@product.route("delete-products/<product_id>", methods=["PATCH"])
```
[the remark this answers is not in the document]
Comment (Freeman-kuch) [reply super_admin_1/products/routes.py:473]: oh, thanks, just read through and actually understand what they are saying

File: super_admin_1/products/routes.py [hunk 19]
```diff
@@ -704,7 +704,7 @@ def permanent_delete(user_id, product_id):
 # Define a route to get all temporarily deleted products
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("/temporarily_deleted_products", methods=["GET"], strict_slashes=False)
+@product.route("/temporarily-deleted-products", methods=["GET"], strict_slashes=False)
```
[the remark this answers is not in the document]
Comment (Freeman-kuch) [reply super_admin_1/products/routes.py:707]: removing the route totally, theres a filter taking care of it

File: super_admin_1/shop/routes.py (hunk 4 above)
Comment (Freeman-kuch) [reply super_admin_1/shop/routes.py:461]: removed the endpoint entirely

**Comment by Freeman-kuch** [comment]
@zxenonx all changes done

**Review by Godhanded — APPROVED** [review]

**Pull request merged**